dbsnp-importer/src/main/java/uk/ac/ebi/eva/dbsnpimporter/io/readers/SampleReader.java

```diff
@@ -111,79 +107,73 @@ public SampleReader(int dbsnpBuild, int batch, String assembly, List<String> ass
-        Map<String, Object> parameterValues = new HashMap<>();
-        parameterValues.put("assemblyType", assemblyTypes);
-        parameterValues.put("assembly", assembly);
-        parameterValues.put("ss_id", getSubsnpId(dbsnpBuild, batch, assembly, assemblyTypes, dataSource));
-        parameterValues.put("batch", batch);
-        return parameterValues;
+    private String buildSql() throws Exception {
+        String sql =
+                "SELECT"
+                        + "    batch.handle AS " + SampleRowMapper.HANDLE
+                        + "   ,batch.batch_id AS " + SampleRowMapper.BATCH_ID
+                        + "   ,batch.loc_batch_id_upp AS " + SampleRowMapper.BATCH_NAME
+                        + "   ,indiv.loc_ind_id_upp AS " + SampleRowMapper.INDIVIDUAL_NAME
+                        + "   ,population.loc_pop_id_upp AS " + SampleRowMapper.POPULATION
+                        + "   ,indiv.ind_id AS " + SampleRowMapper.INDIVIDUAL_ID
+                        + "   ,subind.submitted_ind_id AS " + SampleRowMapper.SUBMITTED_INDIVIDUAL_ID
+                        + "   ,ped.pa_ind_id AS " + SampleRowMapper.FATHER_ID
+                        + "   ,ped.ma_ind_id AS " + SampleRowMapper.MOTHER_ID
+                        + "   ,ped.sex AS " + SampleRowMapper.SEX
+                        + " FROM"
+                        + "    subind "
+                        + "    JOIN submittedindividual indiv on indiv.submitted_ind_id = subind.submitted_ind_id"
+                        + "    LEFT JOIN pedigreeindividual ped on indiv.ind_id = ped.ind_id"
+                        + "    JOIN batch_id_equiv ON batch_id_equiv.subind_batch_id = subind.batch_id"
+                        + "    JOIN batch on batch_id_equiv.subsnp_batch_id = batch.batch_id"
+                        + "    JOIN population on population.pop_id = indiv.pop_id"
+                        + "    JOIN subsnp sub ON subind.subsnp_id = sub.subsnp_id"
+                        + "    JOIN snpsubsnplink link ON sub.subsnp_id = link.subsnp_id"
+                        + "    JOIN b" + dbsnpBuild + "_snpcontigloc loc on loc.snp_id = link.snp_id"
+                        + "    JOIN b" + dbsnpBuild + "_contiginfo ctg ON ctg.contig_gi = loc.ctg_id"
+                        + " WHERE"
+                        + "    sub.subsnp_id = ?"
+                        + "    AND batch.batch_id = ?"
+                        + "    AND ctg.group_term IN (?)"
+                        + "    AND ctg.group_label LIKE ?";
+        return sql;
+    }
+
+    private PreparedStatementSetter buildPreparedStatementSetter() throws Exception {
+        // TODO jmmut: if assemblyTypes has more than one element, this won't work, it seems it is not supported by spring
+        // a solution is allowing a list of static number of elements (AND ctg.group_term IN (?, ?))
+        String joinedAssemblyTypes = String.join(",", assemblyTypes);
```

Comment:
I think the materialized views created for the variants already use a static list consisting of "primary assembly" and "non-nuclear" (or whatever is the capitalization). So we can make this query parameter completely static.